src/main/java/vg/civcraft/mc/civmodcore/serialization/NBTSerialization.java

```diff
@@ -0,0 +1,160 @@
+                    continue;
+                }
+                if (registeredClasses.containsKey(alias)) {
+					System.err.println(errorMessage + "the alias is already registered.");
+                    continue;
+                }
+				System.out.println("NBTSerializable[" + clazz.getName() + "] alias [" + alias + "] registered.");
+                registeredClasses.putIfAbsent(alias, clazz);
+            }
+        }
+    }
+
+    /**
+	 * Unregisters an {@link NBTSerializable} class and all its paths.
+	 *
+	 * @param clazz The NBTSerializable class to unregister.
+	 *
+	 * @throws IllegalArgumentException Throws if the given class to unregister is null.
+	 * */
+    public static <T extends NBTSerializable> void unregisterNBTSerializable(Class<T> clazz) {
+    	if (clazz == null) {
+			throw new IllegalArgumentException("Cannot unregister NBTSerializable: " +
+					"the given class is null.");
+		}
+		System.out.println("NBTSerializable[" + clazz.getName() + "] unregistered.");
+        registeredClasses.values().remove(clazz);
+    }
+
+    /**
+	 * Unregisters all {@link NBTSerializable} classes.
+	 * */
+    public static void clearAllRegistrations() {
+    	System.out.println("All NBTSerialization registered classes have been cleared.");
+    	registeredClasses.clear();
+	}
+
+	/**
+	 * Serializes an {@link NBTSerializable} class into an {@link NBTCompound} for the purpose of being able to
+	 * deserialize it later. For this purpose, {@link NBTSerializable#serialize(NBTCompound) NBTSerializable's
+	 * serialise} method should not be called directly.
+	 *
+	 * @param serializable The {@link NBTSerializable} class to serialize.
+	 * @return Returns an {@link NBTCompound} with the serialised data, or null if the given object was null, or if an
+	 * error occurred during serialization.
+	 * */
+    public static <T extends NBTSerializable> NBTCompound serialize(T serializable) {
+        if (serializable == null) {
+            return null;
+        }
+        NBTCompound nbt = new NBTCompound();
+		try {
+			serializable.serialize(nbt);
+		}
+		catch (Exception exception) {
+			exception.printStackTrace();
+			return null;
+		}
+		nbt.setString(NBT_CLASS_PATH_KEY, serializable.getClass().getName());
+        return nbt;
+    }
+
+    /**
+	 * Deserializes an {@link NBTCompound} into a registered {@link NBTSerializable} class, if it exists.
+	 *
+	 * @param <T> The class type that will be returned. It MUST match the class that was registered exactly.
+	 * @param nbt The NBT data that's used to populate the new instance of the class.
+	 * @return Returns a populated instance of a class, or null. It will return null if 1) the given NBTCompound is null
+	 * or empty, 2) the class identifier cannot be found, 3) the class cannot be matched, or 4) there was an error in
+	 * casting the instance into the given generic type.
+	 * */
+	@SuppressWarnings("unchecked")
+    public static <T extends NBTSerializable> T deserialize(NBTCompound nbt) {
```

Comment:
I think it would better to not make this method generic. You are using the generic as a convenience to cast the result to the desired type, which is not really necessary imo. If the given compound is of a different type than expected by the calling code, then that's a bug in the calling code and it should come up as such there when that code attempts to cast the returned object instead of having CMC swallow that error and return an ambigious `null`. 

So, I'd just return a NBTSerializable here.